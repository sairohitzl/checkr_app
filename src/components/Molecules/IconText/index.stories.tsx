import {ComponentStory,ComponentMeta} from '@storybook/react'

import IconText from '.'

export default{
    title: "Molecules/IconText",
    component: IconText
} as ComponentMeta<typeof IconText>

const Template : ComponentStory<typeof IconText> = (args) => ( <IconText {...args} />)
export const icontext1 = Template.bind({})
icontext1.args = {
    icon:"dashboard",
    title: "Candidates",
}