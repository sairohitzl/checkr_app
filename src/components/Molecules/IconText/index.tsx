import React from 'react'
import { Box } from '@mui/material'
import MyIcon from '../../Atoms/MyIcon'
import Typography from '../../Atoms/Typography'
type Props = {
    icon: string
    title: string
    active?: boolean;
    onClick?: React.MouseEventHandler<HTMLButtonElement>;
}

const IconText = ({icon,title,active,onClick}: Props) => {
  return (
    <Box onClick={()=> onClick} 
      style={{
        padding: "12px",
        background: active? "#EFF2FF" : "#FFFFFF",
        borderRadius: "6px",
        display: "flex",
        justifyContent: "flex-start",
        alignItems: "center"
      }}>
    <MyIcon  icon={icon} colour="#224DFF" ></MyIcon>{" "}
    <span
    style={{
        color: active ? "blue" : "#2C2C2E",
        paddingLeft: "10px",
      }} >
        <Typography variant='body1'>{title}</Typography>
      </span>
    </Box>
  )
}

export default IconText