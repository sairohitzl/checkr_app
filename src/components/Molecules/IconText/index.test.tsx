import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import IconText from ".";
import { render, screen } from "@testing-library/react";

test("Testing IconText Snapshots", () => {
  const tree = renderer
    .create(<IconText icon={"Logout"} title={"John"} active={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
describe("Testing IconText", () => {
  it("render iconText", () => {
    render(<IconText title={"Candidates"} icon={"Copy"} active={true} />);
    const button = screen.getByText("Candidates");
    expect(button).toBeInTheDocument();
    expect(button).toHaveStyle(
      "padding: '12px', background-color: '#EFF2FF', border-radius: '6px'"
    );
  });
});
