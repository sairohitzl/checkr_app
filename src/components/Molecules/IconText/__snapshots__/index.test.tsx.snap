// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing IconText Snapshots 1`] = `
<div
  className="MuiBox-root css-0"
  onClick={[Function]}
  style={
    {
      "alignItems": "center",
      "background": "#EFF2FF",
      "borderRadius": "6px",
      "display": "flex",
      "justifyContent": "flex-start",
      "padding": "12px",
    }
  }
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1308pav-MuiSvgIcon-root"
    data-testid="myicon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M14.12 10.47 12 12.59l-2.13-2.12-1.41 1.41L10.59 14l-2.12 2.12 1.41 1.41L12 15.41l2.12 2.12 1.41-1.41L13.41 14l2.12-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4zM6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9z"
    />
  </svg>
  <span
    style={
      {
        "color": "#3E5FE2",
        "paddingLeft": "10px",
      }
    }
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-1a8qfyx-MuiTypography-root"
    >
      John
    </p>
  </span>
</div>
`;
